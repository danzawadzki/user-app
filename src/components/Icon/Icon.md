Default orange heart icon with all props:

```jsx
<Icon
	name="heart"
	type="far"
	color="orange"
	onClick={() => alert(window.location.href)}
/>
```

Gray sha

```jsx
<Icon
	name="share-square"
	type="fas"
	color="gray"
	onClick={() => alert(window.location.href)}
/>
```
