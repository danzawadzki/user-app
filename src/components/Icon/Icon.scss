@import '../../assets/styles/colors';
@import '../../assets/styles/minor';

.Icon {
	z-index: 1000;
	font-size: 1.25rem;
	transition: $transition--default;
	margin: 0 0.5rem;

	&:hover {
		cursor: pointer;
	}

	&--gray {
		color: $gray--light;

		&:hover {
			color: red;
		}
	}

	&--orange {
		color: $orange;

		&:hover {
			color: $orange;
		}
	}
}
