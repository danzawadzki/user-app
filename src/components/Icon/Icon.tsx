import * as React from 'react';
import './Icon.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconName, IconPrefix, library } from '@fortawesome/fontawesome-svg-core';
import { faHeart } from '@fortawesome/free-regular-svg-icons';
import { faShareSquare } from '@fortawesome/free-solid-svg-icons';

/** Color type */
type color = 'orange' | 'gray';

export interface IIcon {
	/** Icon name */
	name: IconName;
	/** Icon type */
	type: IconPrefix;
	/** Icon color */
	color: color;
	/** Handler to the onClick event */
	onClick: (e: React.MouseEvent<HTMLSpanElement>) => void;
}

/**
 * Registering font-awesome in the library
 *
 * @source https://fontawesome.com/how-to-use/on-the-web/using-with/react
 */
library.add(faHeart, faShareSquare);

/**
 * Icon component
 *
 * @version 1.0.0
 */
const Icon: React.StatelessComponent<IIcon> = ({
	name,
	type = 'fas',
	color,
	onClick
}) => (
	<span className={`Icon Icon--${color}`} onClick={onClick}>
		<FontAwesomeIcon icon={[type, name]} />
	</span>
);

export default Icon;
