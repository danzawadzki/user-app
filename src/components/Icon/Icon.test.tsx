import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';

describe('Icon component: ', () => {
	// Shallow rendered component
	const name = 'heart';
	const type = 'far';
	const color = 'orange';
	const onClick = jest.fn();

	const component = shallow(
		<Icon name={name} type={type} color={color} onClick={onClick} />
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should render correctly', () => {
		expect(component.html()).toContain(name);
		expect(component.html()).toContain(type);
		expect(component.html()).toContain(color);
		expect(component.find('.Icon')).toBeDefined();
	});

	it('should fire onClick', () => {
		expect(onClick).toHaveBeenCalledTimes(0);
		expect(component.find('span.Icon').simulate('click'));
		expect(onClick).toHaveBeenCalledTimes(1);
	});
});
