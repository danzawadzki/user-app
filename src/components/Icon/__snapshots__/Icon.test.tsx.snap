// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component:  should render match snapshot 1`] = `
<span
  className="Icon Icon--orange"
  onClick={[MockFunction]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Array [
        "far",
        "heart",
      ]
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</span>
`;
